docs/net.md

```diff
@@ -0,0 +1,126 @@
+
+```rust
+pub fn distance(node_id_1: H512, node_id_2: H512) -> usize {
+    let hash_1 = Keccak256::digest(node_id_1);
+    let hash_2 = Keccak256::digest(node_id_2);
+    let xor = H256(hash_1.into()) ^ H256(hash_2.into());
+    let distance = U256::from_big_endian(xor.as_bytes());
+    distance.bits().saturating_sub(1)
+}
+```
+
+### Startup
+
+Before starting the server, we do a startup where we connect to an array of seeders or bootnodes. This involves:
+
+- Receiving bootnodes via CLI params
+- Inserting them into our table
+- Pinging them to notify our presence, so they acknowledge us.
+
+This startup is far from being completed. The current state allows us to do basic tests and connections. Later, we want to do a real startup by first trying to connect to those nodes we were previously connected. For that, we'd need to store nodes on the database. If those nodes aren't enough to fill our table, then we also ping some bootnodes, which could be hardcoded or received through the cli. Current issues are opened regarding [startup](https://github.com/lambdaclass/ethereum_rust/issues/398) and [nodes db](DB_LINK).
+
+### Listen loop
+
+The listen loop handles messages sent to our socket. The spec defines 6 types of messages:
+
+- **Ping**: Responds with a `pong` message. If the peer is not in our table we add him, though if already have it filled, then we add it as a replacement for that bucket. If it was inserted we send a `ping from our end to get an endpoint proof.
+- **Pong**: Verifies the `pong` corresponds to a previously sent `ping`, if so we mark the peer as proven..
+- **FindNodes**: Responds with a `neighbors` message that contains as many as the 16 closest nodes from the given target. A target is a pubkey provided by the peer in the message. The response can't be sent in one packet as it might exceed the discv4 max packet size. So we split it into different packets.
+- **Neighbors**: First we verify that we have sent the corresponding `find_node` message. If so, we receive the peers, store them, and ping them.
+- **ENRRequest**: currently not implemented see [here](https://github.com/lambdaclass/ethereum_rust/issues/432).
+- **ENRResponse**: same as above.
+
+### Re-validations
+
+Re-validations are tasks that are implemented as intervals, that is: they run an action every `x` wherever unit of time (currently configured to run every 30 seconds). The current flow of re-validation is as follows
+
+1. Every 30 seconds (by default) we ping the three least recently pinged peers: this may be fine now to keep simplicity, but we might prefer to choose three random peers instead to avoid the search which might become expensive as our buckets start to fill with more peers.
+2. In the next iteration we check if they have answered
+   - if they have: we increment the liveness field by one.
+   - otherwise: we decrement it by the livenes by / 3.
```

Comment:
```suggestion
   - otherwise: we decrement the liveness by a third of its value
```
(This looks confusing, is it liveness = liveness - liveness/3?)